src/hal/tiva_c/spi.rs

```diff
@@ -0,0 +1,304 @@
+  fn write(&self, value: u8) {
+    wait_for!(self.writeable());
+
+    self.regs.ssidr.set_data(value as u16);
+  }
+
+  fn read(&self) -> u8 {
+    wait_for!(self.readable());
+
+    self.regs.ssidr.data() as u8
+  }
+}
+
+#[allow(missing_docs)]
+pub mod reg {
+  //! SSI registers definition
+  use volatile_cell::VolatileCell;
+  use core::ops::Drop;
+
+  ioregs!(Ssi = {
+    /// SSI control register 0
+    0x000 => reg16 ssicr0 {
+      0..3 => dss: rw,    //= Data size select, 0x7 = 8 bits, see datasheet page 970 for other values
+      4..5 => frf: rw,    //= Frame format, 0 = SPI 1 = TI SSF 2 Microwire
+      6    => spo: rw,    //= SSI polarity, 0 = steady staate low on SSInCLK 1 = steady state high
+      7    => sph: rw,    //= SSI clock phase, 1 = first edge 0 = second edge
+      15..8 => scr: rw,   //= SSI clock rate, calculated from formula `SysClk / (CPSDVSR * (1 + SCR))`
+    },
+
+    /// SSI control register 1
+    0x004 => reg16 ssicr1 {
+      0 => lbm: rw,       //= Loopback enable, 1 = enabled 0 = normal operation
+      1 => sse: rw,       //= Serial port enable 1 = enabled 0 = disabled. Must be cleared before SSI can be configured
+      2 => ms: rw,        //= Master/slave select 0 = master 1 = slave
+      4 => eot: rw,       //= End of transmission. Not sure what this is
+    },
+
+    /// SSI send/receive register
+    /// When the SSIDR register is read, the entry in the receive FIFO that is pointed to by the current FIFO read pointer is accessed. When a data value is removed by the SSI receive logic from the incoming data frame, it is placed into the entry in the receive FIFO pointed to by the current FIFO write pointer.
+    /// When the SSIDR register is written to, the entry in the transmit FIFO that is pointed to by the write pointer is written to. Data values are removed from the transmit FIFO one value at a time by the transmit logic. Each data value is loaded into the transmit serial shifter, then serially shifted out onto the SSInTx pin at the programmed bit rate.
+    /// See page 973 in datasheet for more details
+    0x008 => reg16 ssidr {
+      0..15 => data: rw   //= Receive/transmit data
+    },
+
+    /// SSI status register
+    0x00c => reg16 ssisr {
+      0 => tfe: ro,       //= TX FIFO empty 1 = empty 0 = not empty
+      1 => tnf: ro,       //= TX FIFO full 1 = not full 0 = full
+      2 => rne: ro,       //= RX FIFO 1 = not empty 0 = empty
+      3 => rff: ro,       //= RX FIFO full 0 = not full 1 = full
+      4 => bsy: ro,       //= SSI busy 0 = idle, can send data 1 = transmitting, can't send data yet
+    },
+
+    /// SSI clock prescale register
+    0x010 => reg16 ssicpsr {
+      0..7 => cpsdvsr: rw //= SSI clock prescale divisor, 2 – 254 inclusive
+    },
+
+    /// SSI clock source configuration
+    0xfc8 => reg16 ssicc {
+      0..3 => cs: rw,     //= SSI baud clock source, 0x0 = system clock, 0x5 = PIOSC
+    },
+  });
+
+  pub const SSI_0: *const Ssi = 0x40008000 as *const Ssi;
```

Comment:
Please add a TODO to change to placement ioregs. Unfortunately those don't allow multiple addresses, so it would be a mess to use them here now.